backend/core/src/shared/units-transformations.ts

```diff
@@ -32,85 +34,131 @@ const minMaxCurrency = (baseValue: MinMaxCurrency, newValue: number): MinMaxCurr
   }
 }
 
-const bmrHeaders = ["Studio", "1 BR", "2 BR", "3 BR", "4 BR"]
+// Creates data used to display a table of household size/unit size by maximum income per the AMI charts on the units
+// Unit sets can have multiple AMI charts used, in which case the table displays ranges
+const hmiData = (units: Units, maxHouseholdSize: number) => {
+  // Currently, BMR chart is just toggling whether or not the first column shows Household Size or Unit Type
+  const showUnitType = units[0].bmrProgramChart
+
+  type ChartAndPercentage = {
+    percentage: number
+    chart: AmiChart
+  }
+
+  // All unique AMI percentages across all units
+  const allPercentages: number[] = [
+    ...new Set(
+      units.map((unit) => parseInt(unit.amiPercentage, 10)).filter((item) => item != null)
```

Comment:
@emilyjablonski , this looks good, can you just switch the order of the filter and map.